### ISSUE TITLE
Frame unconditional guard over unreachable code

### ISSUE THREAD

**Issue by forthrin** [rubocop/rubocop#15571]
### Before submitting

- [x] I have searched the existing issues and this hasn't been requested yet.

### Cop name

Lint/UnreachableCode

### Is your feature request related to a problem?

Every single line of code following an unconditional branch lights up like a Christmas tree.

> "Take the names of everyone here and bring them in for questioning."
>   — Frank Drebin

### Describe the solution you'd like

Instead of flagging all the poor, innocent subsequent code lines, flag the kingpin:

`Lint/UnconditionalGuard: Code following an always-true guard statement will never run`

### Code examples

```ruby
def main
  return # if 2 + 2 == 4

  _meaning = 42
  puts 'Whatever you say'
  raise PizzaDough if Date.today.friday?
end
```

### Describe alternatives you've considered

Comment out entire code blocks rather than using `return` statements to temporarily disengage code blocks. (Have actually resorted to `return if 2 + 2 == 4` as a moronic workaround to avoid poking highway patrol.)

**Comment by forthrin** [comment]
Computer says:

**Assessment**: The proposal is strong. Lint/UnreachableCode should report the cause rather than every consequence.

**Strategy**: Improve existing `Lint/UnreachableCode` by reporting one offense on the unconditional branch causing unreachable code:

```ruby
return # Lint/UnreachableCode: Code following an unconditional `return` will never run
```

Keep the implementation conservative around:

- `return`, `raise`, `break`, and `next`
- Nested blocks and differing control-flow scopes
- `rescue` and `ensure`
- Conditional branches

Do not autocorrect by removing the branch; it may intentionally suppress a temporary code block. This is a worthwhile diagnostic improvement with low implementation cost.
